### PR TITLE
MES-6751 / Category Development

### PR DESCRIPTION
**Pull request by matthew2564**
## Description
- Port over common NGRX code (Actions / Effects / Reducers)
- Update ported code to be in new NGRX syntax where possible
- Create TestsReducer state and TestReducerFactory logic
- Port over ADI2 specific NGRX
- Change existing usages of 'props' via Actions to use creator functions (https://ngrx.io/api/store/createAction), results in less changes when porting code
- Add tests property to localStorageSyncReducer
- Create placeholder pages i.e. Rekey/Delegated due to NGRX logic being housed there instead of within the modules/store folder

## Checklist

- [x] PR title includes the JIRA ticket number
- [ ] Branch is rebased against the latest develop
- [x] Code has been tested manually
- [x] PR link added to JIRA ticket
- [ ] One review from each scrum team
- [ ] Squashed commit contains the JIRA ticket number

### CODE REVIEW

File: src/app/providers/category-whitelist/category-whitelist.ts [hunk 1]
```diff
@@ -2,7 +2,9 @@ import { TestCategory } from '@dvsa/mes-test-schema/category-definitions/common/
 import { Injectable } from '@angular/core';
 
 // list all test categories that are able to be started by the users
-export const whiteList : TestCategory[] = [];
+export const whiteList : TestCategory[] = [
+  TestCategory.ADI2,
```
Comment (andrewsetterfield) [inline src/app/providers/category-whitelist/category-whitelist.ts:6]: do we want to check this in? if we do we need to remove it before we can release team journals

Comment (matthew2564) [reply src/app/providers/category-whitelist/category-whitelist.ts:6]: Checked in so you can run it to check functionality if there was a desire. We can raise a ticket to remove categories from whitelist prior to Journal/Test Centre Journal release

File: src/store/tests/tests.effects.ts [hunk 2]
```diff
@@ -0,0 +1,449 @@
...
+import { extractTestCentre } from './journal-data/common/test-centre/test-centre.selector';
+import { PopulateTestCategory } from './category/category.actions';
+import { PopulateTestSchemaVersion } from './schema-version/schema-version.actions';
+import { SetExaminerBooked } from './examiner-booked/examiner-booked.actions';
+import { SetExaminerConducted } from './examiner-conducted/examiner-conducted.actions';
+import { SetExaminerKeyed } from './examiner-keyed/examiner-keyed.actions';
+import { MarkAsRekey } from './rekey/rekey.actions';
+// import { getRekeySearchState, RekeySearchModel } from '../../pages/rekey-search/rekey-search.reducer';
+// import { getBookedTestSlot, getStaffNumber } from '../../pages/rekey-search/rekey-search.selector';
+// import { AuthenticationProvider } from '../../providers/authentication/authentication';
```
Comment (Cullima) [inline src/store/tests/tests.effects.ts]: can delete this?

File: src/store/tests/tests.effects.ts [hunk 3]
```diff
@@ -0,0 +1,449 @@
...
+  DelegatedRekeySearchModel,
+  getDelegatedRekeySearchState,
+} from '../../app/pages/delegated-rekey-search/delegated-rekey-search.reducer';
+import { StartTest, TestActionsTypes } from './tests.actions';
+import { createPopulateCandidateDetailsAction } from './journal-data/common/candidate/candidate.action-creator';
+// import { AppResumed, RestartApp } from '../app-info/app-info.actions';
+// import { selectDateConfigLoaded } from '../app-info/app-info.selectors';
+// import { LOGIN_PAGE } from '../../app/pages/page-names.constants';
+
+// import { createPopulateCandidateDetailsAction } from './journal-data/common/candidate/candidate.action-creator';
```
Comment (Cullima) [inline src/store/tests/tests.effects.ts]: delete

File: src/store/tests/tests.effects.ts [hunk 4]
```diff
@@ -0,0 +1,449 @@
...
+import { extractTestSlotAttributes } from './journal-data/common/test-slot-attributes/test-slot-attributes.selector';
+import { PopulateTestCentre } from './journal-data/common/test-centre/test-centre.actions';
+import { extractTestCentre } from './journal-data/common/test-centre/test-centre.selector';
+import { PopulateTestCategory } from './category/category.actions';
+import { PopulateTestSchemaVersion } from './schema-version/schema-version.actions';
+import { SetExaminerBooked } from './examiner-booked/examiner-booked.actions';
+import { SetExaminerConducted } from './examiner-conducted/examiner-conducted.actions';
+import { SetExaminerKeyed } from './examiner-keyed/examiner-keyed.actions';
+import { MarkAsRekey } from './rekey/rekey.actions';
+// import { getRekeySearchState, RekeySearchModel } from '../../pages/rekey-search/rekey-search.reducer';
```
Comment (Cullima) [inline src/store/tests/tests.effects.ts]: think a lot of these commented imports can be deleted

**Review by Cullima — APPROVED** [review]

File: src/store/tests/tests.effects.ts (hunk 2 above)
Comment (matthew2564) [reply src/store/tests/tests.effects.ts]: Will do

File: src/store/tests/tests.effects.ts (hunk 3 above)
Comment (matthew2564) [reply src/store/tests/tests.effects.ts]: Will do

File: src/store/tests/tests.effects.ts (hunk 4 above)
Comment (matthew2564) [reply src/store/tests/tests.effects.ts]: Will do

**Review by tomcrawleyy — APPROVED** [review]

**Review by andrewsetterfield — APPROVED** [review]

**Pull request merged**